src/ex_qdstepandsavetopo.jl

```diff
@@ -1,23 +1,43 @@
 using Revise
+using LinearAlgebra
 using DifferentialEquations
 using JLD2
 using Dates
 using Infiltrator
 using PyPlot
 using Bem2d
 
+# Have access to 500 xeon cores
+# Dedicated access to:
+# * One server has 3TB RAM, 8 P100 GPUs, 80 cores
+# * Two servers have 1TB RAM, 32 cores, 8 V100 GPUs
+
 function derivsconst!(dudt, u, p, t)
-    intidx, nintidx, bemsliptotractotal, els, eta, thetalaw, dc, blockvxglobal, blockvyglobal, dthetadt, dvxdt, dvydt, vx, vy, dtracxglobaldt, dtracyglobaldt, dtracglobaldt = p
-    @views multmatvec!(vx, vy, els.rotmat[intidx, :, :], u[1:3:end], u[2:3:end])
-    @views dtracglobaldt = bemsliptotractotal * [blockvxglobal .- u[1:3:end] blockvyglobal .- u[2:3:end]]'[:]
-    @views multmatvec!(dtracxglobaldt, dtracyglobaldt, els.rotmat[intidx, :, :], dtracglobaldt[1:2:end], dtracglobaldt[2:2:end])
-    for i in 1:nintidx
-        @views dthetadt[i] = thetalaw(abs(vx[i]), abs(u[3:3:end][i]), dc)
-        @views dvxdt[i] = 1 / (eta / els.normalstress[intidx[i]] + els.a[intidx[i]] / abs(vx[i])) * (dtracxglobaldt[i] / els.normalstress[intidx[i]] - els.b[intidx[i]] * dthetadt[i] / u[3:3:end][i])
+    bemsliptotractotal, els, eta, thetalaw, dc, blockvxglobal, blockvyglobal, dthetadt, dvxdt, dvydt, vx, vy, dtracxglobaldt, dtracyglobaldt, dtracglobaldt = p
+    # @views multmatvec!(vx, vy, els.rotmat[intidx, :, :], u[1:3:end], u[2:3:end])
+    @inbounds for i in 1:length(els)
+        vx[i] = els[i].rotmat[1, 1] * u[(1:3:end)[i]] + els[i].rotmat[1, 2] * u[(2:3:end)[i]]
+        vy[i] = els[i].rotmat[2, 1] * u[(1:3:end)[i]] + els[i].rotmat[2, 2] * u[(2:3:end)[i]]
+    end
+
+    # THIS IS WHERE MOST TIME IS SPENT:
+    mul!(dtracglobaldt, bemsliptotractotal, @views([blockvxglobal .- u[1:3:end] blockvyglobal .- u[2:3:end]]')[:])
```

Comment:
This is a very interesting line of code — and is where the vast vast majority of time is spent during integration.